### ISSUE TITLE
Health for RabbitMq is false negative

### ISSUE THREAD

**Issue by YuriiNskyi** [Xabaril/AspNetCore.Diagnostics.HealthChecks#360]
Steps to reproduce:

1. Create ASP NET Core 2.2 project with API template.
2. This is `Startup.cs`:

```
using Microsoft.AspNetCore.Builder;
using Microsoft.AspNetCore.Hosting;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;

namespace CheckHealthChecks
{
    public class Startup
    {
        public Startup(IConfiguration configuration)
        {
            Configuration = configuration;
        }

        public IConfiguration Configuration { get; }

        // This method gets called by the runtime. Use this method to add services to the container.
        public void ConfigureServices(IServiceCollection services)
        {
            services.AddHealthChecksUI();

            services
                .AddHealthChecks()
                .AddRabbitMQ("host=localhost;username=guest;password=guest;publisherConfirms=false;persistentMessages=false");

            services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_2);
        }

        // This method gets called by the runtime. Use this method to configure the HTTP request pipeline.
        public void Configure(IApplicationBuilder app, IHostingEnvironment env)
        {
            if (env.IsDevelopment())
            {
                app.UseDeveloperExceptionPage();
            }
            else
            {
                // The default HSTS value is 30 days. You may want to change this for production scenarios, see https://aka.ms/aspnetcore-hsts.
                app.UseHsts();
            }

            app.UseHttpsRedirection();
            app.UseMvc();
            app.UseHealthChecksUI();
            app.UseHealthChecks("/api/v1/health");
        }
    }
}
```

3. Go to `/api/v1/health`.
4. See the `Unhealthy` response.

I have local `RabbitMq 3.7.7 Erlang 21.2`.

This issue reproduces on every RabbitMq connection string, I tried multiple of them, not only local one.

Why that response is `Unhealthy`? What should be done to supply proper healthchecking?

**Comment by unaizorrilla** [comment]
Hi

Tomorrow morning I try to review this! Sorry for the delay!

**Comment by unaizorrilla** [comment]
Hi @YuriiNskyi 

I review the issue and your code and the problem is related on how you specify the connection string for RabbitMQ.

HealthCheck use URI specification for connection string https://www.rabbitmq.com/uri-spec.html and your are using a wrong format. When the healtcheck try to convert this connection string the format fail and respond your default FailureStatus.

Please use valid RabbitMQ Uri formats!

I close this issue, reopen if you have any other question or this response does not solve your issue!
